python/py-selenium/Portfile

```diff
@@ -4,31 +4,71 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                py-selenium
-version             3.141.0
+version             4.1.0
 license             Apache-2
 maintainers         {dstrubbe @dstrubbe} openmaintainer
 platforms           darwin
 
 description         Python language bindings for Selenium WebDriver
-long_description \
-    The selenium package is used to automate web browser interaction \
-    from Python. Several browsers/drivers are supported (Firefox, \
-    Chrome, Internet Explorer), as well as the Remote protocol.
+long_description    The selenium package is used to automate web \
+                    browser interaction from Python. Several \
+                    browsers/drivers are supported (Firefox, Chrome, \
+                    Internet Explorer), as well as the Remote protocol.
 
 homepage            https://www.seleniumhq.org/
-master_sites        pypi:s/${python.rootname}/
-distname            ${python.rootname}-${version}
 
 python.versions     27 37 38 39 310
 
-checksums           rmd160  6175bf1db6e8de270d0c34b9c58649ab1b118bc7 \
+if {${name} ne ${subport}} {
+    depends_run-append \
+                    port:py${python.version}-urllib3
+
+    if {${python.version} eq 27} {
```

Comment:
Are there other ports that cannot be updated that depend on the 2.7 version? If not, i think we can just drop support. Python 2 is dead.

Reply:
Python2 is EOL, deprecated, and unsupported but also still necessary for many projects. So that makes python2 undead.

I’d prefer not to remove python2 stuff unless necessary.

Reply:
Right, but we package things like Selenium for python 2 because other things in macports that are not yet updated for python 3 still need them. I can't find anything that depends on py-selenium, so that means we have no users of the python 2.7 version, so that means there isn't any reason to still support python 2.7.

Note that we're systematically removing all the python 2.7 things that have no dependents. Whenever we upgrade a "leaf" package to python 3, that often removes the need for remaining packages it depends on to support 2.7 and we recursively update.

Reply:
Thanks, I've removed Python 27.